lib/AST/AutoDiff.cpp

```diff
@@ -261,9 +262,60 @@ static unsigned getNumAutoDiffParameterIndices(AnyFunctionType *fnTy) {
   return numAutoDiffParameterIndices;
 }
 
+/// Returns true if the given type conforms to `Differentiable` in the given
+/// module.
+static bool conformsToDifferentiableInModule(Type type, ModuleDecl *module) {
```

Comment:
Note: `conformsToDifferentiableInModule` is duped in `TypeCheckAttr.cpp`. It could be declared in `AutoDiff.h` to eliminate dupe, holding off for now though.